README.md

```diff
@@ -54,12 +57,173 @@ with an appropriate schema.
+
+_TODO:_ Should we bake all these in a docker container to make it easier?
+
+##### Build the latest VectorPipe
+
+- Clone https://github.com/geotrellis/vectorpipe
+- `cd vectorpipe`
+- `sbt publish-local`
+
+### Building and running project services
+
+Rebuild Docker images and run the services.
+
+```sh
+vagrant up
+vagrant ssh
+./scripts/update.sh
+./scripts/services.sh
+```
+
+### Ports
+
+| Service            | Port                               |
+| ------------------ | ---------------------------------- |
+| HBase UI           | [`16010`](http://localhost:16010/) |
+(TODO)
+
+### Testing (TODO)
+
+```
+./scripts/test.sh
+```
+
+### Scripts
+
+| Name           | Description                                                   |
+| -------------- | ------------------------------------------------------------- |
+| `cibuild.sh`   | Build project for CI (TODO)                                   |
+| `clean.sh`     | Free disk space by cleaning up dangling Docker images         |
+| `console.sh`   | Run interactive shell inside application container            |
+| `geoserver.sh` | Run GeoServer in the dev network                              |
+| `sbt.sh`       | Run an sbt console in the Spark container in the dev network  |
+| `services.sh`  | Run Docker Compose services                                   |
+| `setup.sh`     | Provision Vagrant VM and run `update.sh`                      |
+| `test.sh`      | Run unit tests       (TODO)                                   |
+| `update.sh`    | Build Docker images                                           |
+
+### Docker setup
+
+![Docker Dev setup](docker-dev.svg)
+
+The development environment contains a number of docker-compose files which are used to
+run local development versions of the various services that a deployment environment would have,
+as  the update/query containers.
+Spark jobs will be run by scripts, and bring up a docker container that connects to the default network.
+The docker-compose.services.yml is the only one that brings up a docker network, which is named
+`osmesa_default`. All other containers need to connect to this network in order to communicate
+with the other components.
+
+Below is a description of the various docker-compose files and scripts to run spark jobs:
+
+| Name                            | Description                                           |
+| ------------------------------- | ----------------------------------------------------- |
+| `docker-compose.services.yml`   | Development services, including GeoMesa-enabled HBase |
+| `docker-compose.geoserver.yml`  | Brings up a GeoServer instance in dev network         |
+| `docker-compose.sbt.yml`        | Runs a sbt console in the spark-enabled container     |
+| `docker-compose.update.yml`     | Container that runs the OSMesa update service [TODO]  |
+| `docker-compose.query.yml`      | Container that runs the OSMesa query service  [TODO]  |
+
+#### GeoServer
```

Comment:
James recently [added](https://github.com/PublicMapping/DistrictBuilder/pull/464) GeoServer to the development environment of DistrictBuilder. He took a different approach (used Jetty) vs. Tomcat. I don't really have a good sense for which is better yet, but I suspect the Jetty approach will be lower overhead.